### PR TITLE
deliver packet not msg from session

### PR DESCRIPTION
**Pull request by snichme**
`Session` delivers `MQTT:Publish` instead of `Envelope` to consumer

### CODE REVIEW

File: src/lavinmq/mqtt/session.cr [hunk 1]
```diff
@@ -131,6 +133,20 @@ module LavinMQ
...
+        msg = env.message
+        retained = msg.properties.try &.headers.try &.["x-mqtt-retain"]? == true
+        qos = msg.properties.delivery_mode || 0u8
+        MQTT::Publish.new(
+          packet_id: packet_id,
+          payload: msg.body,
+          dup: env.redelivered,
+          qos: qos,
+          retain: retained,
+          topic: msg.routing_key.tr(".", "/")
```
Comment (snichme) [inline src/lavinmq/mqtt/session.cr:146]: @spuun is this `.tr`  really correct?

Comment (spuun) [reply src/lavinmq/mqtt/session.cr:146]: I don't think you need to between topic and routing key here. I think we're storing it as topic. 
But if it's a routing key you want to convert `.` to `/`.

Comment (kickster97) [reply src/lavinmq/mqtt/session.cr:146]: yep thats a rest that i forgot to remove

Comment (snichme) [reply src/lavinmq/mqtt/session.cr]: removing the `tr` then. 

WDYT of the change otherwise?

**Review by kickster97 — APPROVED** [review]

File: src/lavinmq/mqtt/session.cr (hunk 1 above)
Comment (kickster97) [reply src/lavinmq/mqtt/session.cr]: looks good :)

**Pull request merged**